app/controllers/hashtags_controller.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class HashtagsController < ApplicationController
+  requires_login
+
+  def show
+    raise Discourse::InvalidParameters.new(:slugs) if !params[:slugs].is_a?(Array)
+
+    all_slugs = []
+    tag_slugs = []
+
+    params[:slugs].each do |slug|
+      if slug.end_with?(PrettyText::Helpers::TAG_HASHTAG_POSTFIX)
+        tag_slugs << slug.chomp(PrettyText::Helpers::TAG_HASHTAG_POSTFIX)
+      else
+        all_slugs << slug
+      end
+    end
+
+    # Try to resolve hashtags as categories first
+    category_slugs_and_ids = all_slugs.map { |slug| [slug, Category.query_from_hashtag_slug(slug)&.id] }.to_h
+    categories_by_id = Category.secured(guardian).where(id: category_slugs_and_ids.values).map { |c| [c.id, c] }.to_h
```

Comment:
Do we need to load the whole `Category` object just to return the `url` later on?

Reply:
Good point.